hibernate-core/src/main/java/org/hibernate/internal/util/NullnessHelper.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.internal.util;
+
+import java.util.function.Supplier;
+
+/**
+ * @author Steve Ebersole
+ */
+public class NullnessHelper {
+
+	/**
+	 * Operates like SQL coalesce expression, except empty strings are treated as null.  Return the first non-empty value
+	 *
+	 * @param values The list of values.
+	 * @param <T> Generic type of values to coalesce
+	 *
+	 * @return The first non-empty value, or null if all values were empty
+	 */
+	public static <T> T coalesce(T... values) {
+		if ( values == null ) {
+			return null;
+		}
+		for ( T value : values ) {
+			if ( value != null ) {
+				if ( String.class.isInstance( value ) ) {
+					if ( StringHelper.isNotEmpty( (String) value ) ) {
```

Comment:
Minor nitpicking, as I've recently being removing the use of "unnecessary helpers":

when you know already that "value" is not null, there is no reason to use the `StringHelper`. Just go with `((String) value).isEmpty()` ?

Reply:
Thanks @Sanne , I'll change it.

Reply:
I guess you missed this one then - I literally copied this as is :D

But, +1